### PR TITLE
feat: compatibility with browser usage

### PR DESCRIPTION
**Pull request by JounQin**

### CODE REVIEW

File: _js/go.mod [hunk 1]
```diff
@@ -8,5 +8,6 @@ replace mvdan.cc/sh/v3 => ../
 
 require (
 	github.com/gopherjs/gopherjs v0.0.0-00010101000000-000000000000
+	golang.org/dl v0.0.0-20200811212135-d149fc5456ff // indirect
```
Comment (mvdan) [inline _js/go.mod:11]: undo this, and all the go.sum changes.

File: _js/build [hunk 2]
```diff
@@ -1,12 +1,13 @@
 #!/bin/sh
 
-if ! go version | grep -q go1.12; then
+if ! command -v "go1.12" >/dev/null 2>&1; then
 	echo "Go 1.12.x required to build."
+	echo "run \`go get golang.org/dl/go1.12 && go1.12 download\` first"
```
Comment (mvdan) [inline _js/build:5]: undo this as well. you can set up your PATH accordingly instead, like:

```
mkdir -p some/bin
ln -sf $(which go1.12) some/bin/go
export PATH="$PWD/some/bin:$PATH"
# run your command here
```

Most people will likely use the 1.12 wrapper, but we shouldn't add a hard dependency on it.

File: _js/index.html [hunk 3]
```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
```
Comment (mvdan) [inline _js/index.html:1]: I'm not sure I understand what this file does. Please document it.

File: _js/main.go [hunk 4]
```diff
@@ -14,7 +14,19 @@ import (
...
+		exps = js.Module.Get("exports")
+	} else {
+		exps = js.MakeWrapper(map[string]interface{}{})
+		originalGlobalSh := js.Global.Get("sh")
+		exps.Set("noConflict", func() *js.Object {
+			js.Global.Set("sh", originalGlobalSh)
+			return exps
+		})
+		js.Global.Set("sh", exps)
+	}
```
Comment (mvdan) [inline _js/main.go:29]: I also don't understand what any of this does, sorry - please document it.

**Review by mvdan — CHANGES_REQUESTED** [review]
I can imagine what you're trying to do here, but without any comments and no commit message, I just don't understand much. Please expand via comments.

File: _js/build (hunk 2 above)
Comment (JounQin) [reply _js/build:5]: The main go package `Requires Go 1.14 or later`, so I imagined that we have to install multiple go versions, but if you don't like it, I can revert it.

File: _js/go.mod (hunk 1 above)
Comment (JounQin) [reply _js/go.mod:11]: OK, it should be related to `go get golang.org/dl/go1.12`.

File: _js/index.html (hunk 3 above)
Comment (JounQin) [reply _js/index.html:1]: Testing browser usage, or should I rename this file?

File: _js/main.go (hunk 4 above)
Comment (JounQin) [reply _js/main.go:29]: In this condition, `module` is not available, so we need to assign `window.sh` to get access to exported `sh` object outside.

File: _js/build (hunk 2 above)
Comment (mvdan) [reply _js/build:5]: Yes, you will need to install multiple versions unfortunately, but what I'm saying is that you can do that outside the script pretty easily without having to hard-code it into the script :)

**Comment by JounQin** [comment]
> I can imagine what you're trying to do here, but without any comments and no commit message, I just don't understand much. Please expand via comments.

@mvdan `js.Module.Get('exports')` means `module.exports` but it is not available on browser, so we need to check whether `module` be available. On browser, we need to set the reference to `window.sh`(or something else).

**Comment by mvdan** [comment]
Gotcha, the explanations above make sense. Just leave them as comments in the code, because I want them in git.

**Comment by JounQin** [comment]
@mvdan Done.

File: _js/build (hunk 2 above)
Comment (JounQin) [reply _js/build]: It seems redundant to change go versions between folders at a same project personally, but of course it's fine to keep it as-is.

**Review by mvdan — APPROVED** [review]
Thank you!

**Pull request merged**